release.py

```diff
@@ -0,0 +1,440 @@
+import json
+import os
+import re
+import subprocess
+
+try:
+    from setuptools import Command
+except ImportError:
+    from distutils.core import Command
+
+REPOSITORY_API = 'https://api.github.com/repos/zachyee/presto-admin'
+CURRENT_DIRECTORY = os.path.dirname(os.path.realpath(__file__))
+RELEASE_LIST_DOC_PATH = os.path.join(CURRENT_DIRECTORY, 'docs/release.rst')
+RELEASE_DOCS_DIRECTORY = os.path.join(CURRENT_DIRECTORY, 'docs/release/')
+
+
+class release(Command):
+    description = 'create source release to github and/or pypi'
+
+    user_options = [('github', None,
+                     'boolean flag indicating if a release should be created for github'),
+                    ('pypi', None,
+                     'boolean flag indicating if a release should be created for pypi'),
+                    ('all', None,
+                     'boolean flag indicating if a release should be created for github and pypi')]
+
+    @staticmethod
+    def get_latest_release():
+        latest_release = subprocess.check_output(['curl', '--silent', REPOSITORY_API + '/releases/latest'])
+        return json.loads(latest_release)
+
+    @staticmethod
+    def get_remote_branches():
+        branches = subprocess.check_output(['curl', '--silent', REPOSITORY_API + '/branches'])
+        return json.loads(branches)
+
+    @staticmethod
+    def check_branch_remote_exists(local_branch):
+        for remote_branch in release.get_remote_branches():
+            if local_branch == remote_branch['name']:
+                print 'Local branch %s exists remotely' % local_branch
+                return
+        exit('Local branch %s does not exist remotely.' % local_branch)
+
+    @staticmethod
+    def check_branch_up_to_date(local_branch):
+        status = subprocess.check_output(['git', 'status'])
```

Comment:
See https://github.com/kokosing/docker-release/blob/master/docker_release/main.py for an example